File: Cryptography/Caesar Cipher/without-header.cpp
#include <iostream>
using namespace std;
 
// This function receives text and shift and
// returns the encrypted text
string encrypt(string text, int distance)
{
    string result = "";
 
    // traverse text
    for (int i = 0; i < text.length(); i++) {
        // apply transformation to each character
        // Encrypt Uppercase letters
        if (isupper(text[i]))
            result += char(int(text[i] + distance - 65) % 26 + 65);
 
        // Encrypt Lowercase letters
        else
            result += char(int(text[i] + distance - 97) % 26 + 97);
    }

    // Return the resulting string
    return result;
}

// Driver program to test the above function
int main()
{
    string text = "You Look Lonely";
    int distance = 3;
    cout << "Text : " << text;
    cout << "\nShift: " << distance;
    cout << "\nCipher: " << encrypt(text, distance);
    return 0;
}

File: Cryptography/Caesar Cipher/without-header.go
package main

import "fmt"

func encrypt(text string, distance int) string {
	result := ""

	for _, char := range text {
		if char >= 'A' && char <= 'Z' {
			encrypted := ((int(char) - int('A') + distance) % 26) + int('A')
			result += string(rune(encrypted))
		} else if char >= 'a' && char <= 'z' {
			encrypted := ((int(char) - int('a') + distance) % 26) + int('a')
			result += string(rune(encrypted))
		} else {
			result += string(char)
		}
	}

	return result
}

func main() {
	text := "You Look Lonely"
	distance := 3
	fmt.Println("Text:", text)
	fmt.Println("Shift:", distance)
	fmt.Println("Cipher:", encrypt(text, distance))
}
